File: cpp/Loader.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include "Loader.h"

std::vector<int> Loader::Load(std::string path)
{
    std::vector<int> vector;
    std::ifstream file;

    file.open(path);

    if(!file)
    {
        std::cerr << "Error while opening the file " << path << std::endl;
        return std::vector<int>();
    }
    
    int number;
    while(file >> number)
    {
        vector.push_back(number);
    }
    file.close();
    // std::cout << "Vector length: " << vector.size() << std::endl;

    return vector;
}

File: python/loader.py
def print_content(path: str):
    file = open(path, "r")
    content = file.read()
    file.close()
    print(content)


def load_to_array(path:str) -> []:
    result = []
    line = ""
    number = 0
    file = open(path, "r")
    for line in file:
        number = int(line)
        result.append(number)

    file.close()
    return result

